packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -100,11 +90,6 @@ export class UPlotChart extends Component<PlotProps, UPlotChartState> {
       this.reinitPlot();
     } else if (!sameData(prevProps, this.props)) {
       plot?.setData(this.props.data as AlignedData);
-    } else if (!sameTimeRange(prevProps, this.props)) {
-      plot?.setScale('x', {
-        min: this.props.timeRange.from.valueOf(),
-        max: this.props.timeRange.to.valueOf(),
-      });
```

Comment:
this is the bit that was causing the problem, since the `timeRange` changed before the `.setData()` fired (above), since the time picker change triggers re-render before queries return with new data.